src/Models/NonhydrostaticModels/boundary_mass_fluxes.jl

```diff
@@ -0,0 +1,239 @@
+    end
+
+    # Check top boundary (w velocity)
+    top_flux_and_area = initialize_boundary_mass_flux(w, w_bcs.top, Val(:top))
+    boundary_fluxes = merge(boundary_fluxes, top_flux_and_area)
+    if needs_mass_flux_correction(w_bcs.top)
+        push!(right_matching_scheme_boundaries, :top)
+        total_area_matching_scheme_boundaries += boundary_fluxes.top_area
+    end
+
+    boundary_fluxes = merge(boundary_fluxes, (; left_matching_scheme_boundaries,
+                                                right_matching_scheme_boundaries,
+                                                total_area_matching_scheme_boundaries))
+
+    return boundary_fluxes
+end
+
+update_open_boundary_mass_fluxes!(model) = map(compute!, model.boundary_mass_fluxes)
+
+open_boundary_mass_flux(model, bc::OBC, ::Val{:west}, u) = model.boundary_mass_fluxes.west_mass_flux |> maximum
+open_boundary_mass_flux(model, bc::OBC, ::Val{:east}, u) = model.boundary_mass_fluxes.east_mass_flux |> maximum
+open_boundary_mass_flux(model, bc::OBC, ::Val{:south}, v) = model.boundary_mass_fluxes.south_mass_flux |> maximum
+open_boundary_mass_flux(model, bc::OBC, ::Val{:north}, v) = model.boundary_mass_fluxes.north_mass_flux |> maximum
+open_boundary_mass_flux(model, bc::OBC, ::Val{:bottom}, w) = model.boundary_mass_fluxes.bottom_mass_flux |> maximum
+open_boundary_mass_flux(model, bc::OBC, ::Val{:top}, w) = model.boundary_mass_fluxes.top_mass_flux |> maximum
+
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:west}, u) = zero(model.grid)
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:east}, u) = zero(model.grid)
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:south}, v) = zero(model.grid)
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:north}, v) = zero(model.grid)
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:bottom}, w) = zero(model.grid)
+open_boundary_mass_flux(model, bc::ZIOBC, ::Val{:top}, w) = zero(model.grid)
+
+open_boundary_mass_flux(model, bc, side, velocity) = zero(model.grid)
+
+function open_boundary_mass_inflow(model)
+    update_open_boundary_mass_fluxes!(model)
+
+    u, v, w = model.velocities
+    u_bcs = u.boundary_conditions
+    v_bcs = v.boundary_conditions
+    w_bcs = w.boundary_conditions
+
+    total_flux = zero(model.grid)
+
+    # Add flux through left boundaries
+    total_flux += open_boundary_mass_flux(model, u_bcs.west, Val(:west), u)
+    total_flux += open_boundary_mass_flux(model, v_bcs.south, Val(:south), v)
+    total_flux += open_boundary_mass_flux(model, w_bcs.bottom, Val(:bottom), w)
+
+    # Subtract flux through right boundaries
+    total_flux -= open_boundary_mass_flux(model, u_bcs.east, Val(:east), u)
+    total_flux -= open_boundary_mass_flux(model, v_bcs.north, Val(:north), v)
+    total_flux -= open_boundary_mass_flux(model, w_bcs.top, Val(:top), w)
+
+    return total_flux
+end
+
+correct_left_boundary_mass_flux!(u, bc::OBC, ::Val{:west},   avg_mass_inflow) = interior(u, 1, :, :) .-= avg_mass_inflow
```

Comment:
here and in some other places, I think defining a mathematical notation will help make this easier to read: eg `δu` instead of `avg_mass_inflow`. But I am ok either way

Reply:
I preferred to use `*_inflow` in some of these names because it makes it clear that a positive value means added mass to the domain. Otherwise this could be a source of confusion I think.